### PR TITLE
experimenting with codegen for this

### PR DESCRIPTION
**Pull request by bruce-lindsay**

### CODE REVIEW

**Comment by bruce-lindsay** [comment]
@jamessimone not fully happy with this, but it does _work._  Just syncing up to origin in case you're interested in looking at it.

**Comment by bruce-lindsay** [comment]
next i'll probably add an option to generate the file for inheritance since in the case of string it's only a partial implementation because it doesn't contain all the nice string specific stuff you've already done. Also I image string will have shouldMatch(Pattern p) etc

File: src/classes/ExpectDatetime.cls [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    public ExpectDatetime(Datetime expected) {
+        this.expected = expected;
+    }
+    private ExpectDatetime assertThis(Boolean test, String message) {
+        Expect.assert(test, message);
+        return this;
+    }
+    public ExpectDatetime shouldEqual(Datetime actual) {
+        return shouldEqual(actual, String.format(DATETIME_SHOULD_BE_MESSAGE, new String[]{
```
Comment (jamessimone) [inline src/classes/ExpectDatetime.cls]: I think these String.format calls can be re-factored into the base Expect class with the formatMessage method I have in my branch?  I think it's nicer to have the static strings consolidated - your thoughts?

Comment (bruce-lindsay) [reply src/classes/ExpectDatetime.cls]: Yeah I agree. I was just focused on getting the templating proof of concept working. I can see how it's not ideal yet though

**Comment by jamessimone** [comment]
I think the template in general looks good!  but ideally, yeah, I would like to sync up our branches or at least retain some of the more specialized methods (particularly for String and SObject, since those are probably the most specialized).

**Review by jamessimone — APPROVED** [review]

**Comment by bruce-lindsay** [comment]
needs more work

**Pull request closed without merging**